File: java/ex3_1_2.java
class Chooser {
    <T> T choose(T v1, T v2) { return true ? v1 : v2; }
}

class Example {
    public static void main(String[] args) {
        Chooser c = new Chooser();
        c.<Integer>choose(1, 2);
    }
}

File: rust/ex3_1_2.rs
struct Chooser;

impl Chooser {
    fn choose<T>(&self, v1: T, v2: T) -> T { if true { v1 } else { v2 } }
}

fn main() {
    let c = Chooser;
    c.choose::<i32>(1, 2);
}
